### ISSUE TITLE
Make compatible with jest 23

### ISSUE THREAD

**Issue by floschnell** [thymikee/jest-preset-angular#160]
When using jest 23.x the package resolver (yarn in my case) will show `warning " > jest-preset-angular@5.2.2" has incorrect peer dependency "jest@^22.0.0".`
I can still use that new version, I'm unsure if there are any breaking changes or whether we can just loosen those restrictions.

**Comment by thymikee** [comment]
There were couple of breaking changes in 23, but not affecting this preset too much AFACT. We're waiting for `ts-jest` to upgrade to 23 though.

**Comment by JLHwung** [comment]
@thymikee `ts-jest` has been updated to support jest 23: https://github.com/kulshekhar/ts-jest/commit/52145c71b43b44d5bc05c72fe296d2ee6909a2fc

**Comment by ahnpnl** [comment]
@JLHwung unfortunately they haven't released their master branch yet.

**Comment by davinkevin** [comment]
Waiting for this one for my schematics (https://github.com/davinkevin/jest)  allowing simple configuration of `jest` inside of an Angular CLI project with the help of this project

**Comment by paulsouche** [comment]
I tried to update to jest 23 and I've got an error `TypeError Event is not a constructor` and indeed global Event is null ? Is it in the couple of breaking changes ? I can't find anything related in Jest Changelog ?

**Comment by ahnpnl** [comment]
Hi,

Unfortunately we aren't sure about supporting Jest 23 fully because we have a dependency on ts-jest. ts-jest hasn't updated to support Jest 23 yet so we can't test your case now.

**Comment by paulsouche** [comment]
Thanks @ahnpnl, I found a workaround anyway. I'll keep that in mind for when everything will be ok. Thanks again for the great job on that preset !

**Comment by blackholegalaxy** [comment]
@paulsouche could you share your solution? We ran into the same issue.

**Comment by paulsouche** [comment]
@blackholegalaxy Event mock polyfill

```typescript
class Event {
  preventDefault = jest.fn();
  stopPropagation = jest.fn();
  constructor(meta: any = {}) {
    Object.assign(this, meta);
  }
}

// in your config
global.Event = Event;
```

**Comment by squelix** [comment]
`ts-jest` has been released in version 23.0.0 :)

**Comment by thymikee** [comment]
Thanks @squelix, already fixed by: https://github.com/thymikee/jest-preset-angular/pull/166

**Comment by thymikee** [comment]
Just published `next` tag being v6.0.0-rc.0, feel free to give it a try and later this week I'll release v6 ;)

**Comment by blackholegalaxy** [comment]
Thanks @thymikee
